app/src/main/java/com/example/pocketwatching/Etc/TokenSorter.java

```diff
@@ -0,0 +1,161 @@
+package com.example.pocketwatching.Etc;
+
+import android.util.Log;
+
+import com.example.pocketwatching.Models.Ethplorer.PortfolioValues.Price;
+import com.example.pocketwatching.Models.Ethplorer.PortfolioValues.Token;
+
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+
+public class TokenSorter {
+    private String type;
+    private Boolean descending;
+    private List<Token> tokens;
+
+    public TokenSorter(List<Token> tokens, String type, Boolean descending) {
+        this.tokens = tokens;
+        this.type = type;
+        this.descending = descending;
+    }
+
+    public void sort() {
+        int start = 0;
+        int end = tokens.size() - 1;
+
+        // add alpha sort case here
+        numSort(start, end);
+
+        if (descending == true) {
+            reverseList(tokens);
+        }
+    }
+
+    public void numSort(int start, int end) {
+        if ((start < end) && ((end - start) >= 1)) {
+            int mid = (end + start) / 2;
+            numSort(start, mid);
+            numSort(mid + 1, end);
+
+            switch (type) {
+                case "Balance":
+                    balanceSort(start, mid, end);
+                    break;
+                case "Name":
+                    break;
+                case "Symbol":
+                    break;
+                case "Percent Change (24h)":
+                    pctSort(start, mid, end);
+                    break;
+                default:
+                    balanceSort(start, mid, end);
```

Comment:
I know we talked earlier about generalizing this switch statement to one or two functions and usage of comparators, but as a quick tidbit for cleaner, not as repetitive code, since the control flow for case "Balance" and default are the same, you can move the "Balance" case right above the default case and let it fall through to default like:
```
case "Balance":
default:
    balanceSort(start, mid, end)
```
This works since there's no ```break``` call in the "Balance" case, so it'll run through subsequent cases until you finish the default case or the next ```break``` keyword.